Add library app design and logic

descholar-ceo opened the pull request:
## What does this PR do?
It adds the library app design and logic and it closes [#2](https://github.com/descholar-ceo/library/issues/2)

## Tasks
- [x] Project initialization
- [x] Add the User interface and design
- [x] Add some logic in the js file
- [x] Polish codes and apply the best practices
- [x] Fix all eslint and stylelint errors
- [x] Add professional readme file

## Deployment
This branch is deployed on [this link](https://quizzical-lumiere-418bd9.netlify.app/)

descholar-ceo committed:
Project initialization

descholar-ceo committed:
Add form for creating a new book

descholar-ceo committed:
Style the form to look better

descholar-ceo committed:
Fix stylelint errors

Bluette1 committed:
Fix positioning of radio inputs, add margins and paddings in form

Bluette1 committed:
Add some logic for adding a book to the library

descholar-ceo committed:
Add display books

Bluette1 committed:
Add more styling

Bluette1 committed:
Add delete book and update read status actions

Bluette1 committed:
Add delete book and update read status actions

descholar-ceo committed:
Add professional readme file and add the local storage to persist book

Dandush03 reviewed (changes requested):
# Status: :red_circle: Request Changes :red_circle: 
### Great work with this project, I will add some changes you must do in order to get approved
### MANDATORY
- [x] __ES6 Syntax:__ You most be using ES6 syntax which means you need to change your current functions to an arrow function
- [x] __Separate Logic From Dom:__ Keep the application logic separated from DOM manipulation tasks

### OPTIONAL

- [x] __README:__  You don't need to run npm with this project since everything is right there. You only need to clone the readme and double click the file in order to run it

## That's All. Happy Coding :tophat: :computer:

Bluette1 committed:
Highlight only missing fields

Bluette1 committed:
Fix lint errors

Bluette1 committed:
Fix readme, add es6 arrow syntax for functions

descholar-ceo committed:
Separate logic from DOM manipulations

descholar-ceo commented:
> # Status:  Request Changes 
> ### Great work with this project, I will add some changes you must do in order to get approved
> ### MANDATORY
> * [x]  **ES6 Syntax:** You most be using ES6 syntax which means you need to change your current functions to an arrow function
> * [x]  **Separate Logic From Dom:** Keep the application logic separated from DOM manipulation tasks
> 
> ### OPTIONAL
> * [x]  **README:**  You don't need to run npm with this project since everything is right there. You only need to clone the readme and double click the file in order to run it
> 
> ## That's All. Happy Coding

@Dandush03 Thank you so much for this review, we have implemented all the `MANDATORY` changes requested and implemented the `OPTIONAL` requirements as well!

lekegitrepo reviewed (approved):
## Status: Approved :heavy_check_mark:
### You have done a good job :+1: by implementing the required changes. The project is approved :white_check_mark:

#### _Reviewed by Temitayo Adeleke._
